### ISSUE TITLE
kudos plugin itself does not depend on jquery-cookie plugin

### ISSUE THREAD

**Issue by amitu** [masukomi/kudos#11]
The demo application/index.html depends on jquery-cookie plugin, but not the kudos.js plugin itself, yet its mentioned in dependencies that kudos depends on cookie plugin.

**Comment by masukomi** [comment]
Yes, and no....

index.html line 20

```
if($.cookie(postId) == 'true') {
```

The thing is, that while you can run kudos.js without cookies, doing so  will enable people to give an additional kudo to the same posts/articles/things every time they load the page. You HAVE to keep track of the user _with_ cookies if you want any hope of of having a realistic count. the idea here is not to get people to click it as many times as possible. The idea is to have a count of how many people have not just read, but appreciated something you have created. 

So, technically you're right, it does not depend on jquery-cookie plugin, but it does require jquery, and if you are using jquery and need to use cookies, then it's the most logical choice. I may tweak the docs to suggest that it needs to track via cookies, and that that plugin is suggested.

As an aside, it should be noted that kudos.js is _not_ a jquery plugin. It's just a simple JavaScript library. 

I appreciate the feedback. I just want to make sure you understand that this is really not going to function correctly if you don't track users with cookies.

**Comment by amitu** [comment]
I just implemented your plugin on my [jekyll powered blog](http://amitu.com/2013/04/kudos-using-parse-for-jekyll/), and I totally love it :-)

In my case I used jStorage instead of cookie. Cookie would be a very bad choice as if there are a lot of posts, the cookie will not work, due to their size limit. Also cookies are sent with every request back to server, which is a wastage. 

I totally agree that some kind of local store would be required, if not a server side database, just that cookies are 1. not technically a requirement for your awesome plugin and 2. a bad choice to begin with.

**Comment by masukomi** [comment]
yes, I'm not suggesting that storing a cookie for every one is the right solution, or that trying to shove info on all kudos into one cookies is the right solution but at least one cookies does need to be used and the cookie plugin is not a bad solution . I guess the docs should be updated to note something to that effect.

HOWEVER jStore may actually be a bad choice because it uses local storage. LocalStorage (used by jStore) has no guarantee of permanence, and CAN be blown away by the browser at any time if space is needed. For example, iOS blows away data in LocalStorage fairly regularly.

**Comment by amitu** [comment]
I did not know about that last aspect. Mozilla docs seem to indicate localstorage expiry should be treated the same way cookie expiry is, but cookies have explicit expiry and I am not sure about localstorage. 

Given that I am using Parse, I can use their [AnonymousUserUtils](https://www.parse.com/docs/android/api/com/parse/ParseAnonymousUtils.html) to store data on Parse instead of localstorage. 

Eventually, as you have mentioned in the example somewhere, if we really care about persistence, server side code will be needed.

The purpose of this "issue" was a technicality that, despite whatever else we discussed, the plugin does not actually depend on $.cookie. I know its a very minor nit-pic.
